From 2bdbca2c39cd0947b84bc2e2f4d86938864625fd Mon Sep 17 00:00:00 2001
From: David Vernet <void@manifault.com>
Date: Wed, 10 Jan 2024 12:39:33 -0600
Subject: [PATCH 094/130] scx: Copy scx_exit_kind to scx_test.h

Rather than define the error value in each test, let's just define it in
scx_test.h.

Signed-off-by: David Vernet <void@manifault.com>
---
 tools/testing/selftests/scx/ddsp_bogus_dsq_fail.c   |  2 --
 tools/testing/selftests/scx/ddsp_vtimelocal_fail.c  |  2 --
 tools/testing/selftests/scx/scx_test.h              | 13 +++++++++++++
 .../selftests/scx/select_cpu_dispatch_bad_dsq.c     |  2 --
 .../selftests/scx/select_cpu_dispatch_dbl_dsp.c     |  2 --
 5 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/tools/testing/selftests/scx/ddsp_bogus_dsq_fail.c b/tools/testing/selftests/scx/ddsp_bogus_dsq_fail.c
index 0310fa58f..ef8ee04ff 100644
--- a/tools/testing/selftests/scx/ddsp_bogus_dsq_fail.c
+++ b/tools/testing/selftests/scx/ddsp_bogus_dsq_fail.c
@@ -11,8 +11,6 @@
 #include "ddsp_bogus_dsq_fail.bpf.skel.h"
 #include "scx_test.h"
 
-#define SCX_EXIT_ERROR 1024
-
 static enum scx_test_status setup(void **ctx)
 {
 	struct ddsp_bogus_dsq_fail *skel;
diff --git a/tools/testing/selftests/scx/ddsp_vtimelocal_fail.c b/tools/testing/selftests/scx/ddsp_vtimelocal_fail.c
index 211e84886..b55611cd0 100644
--- a/tools/testing/selftests/scx/ddsp_vtimelocal_fail.c
+++ b/tools/testing/selftests/scx/ddsp_vtimelocal_fail.c
@@ -10,8 +10,6 @@
 #include "ddsp_vtimelocal_fail.bpf.skel.h"
 #include "scx_test.h"
 
-#define SCX_EXIT_ERROR 1024
-
 static enum scx_test_status setup(void **ctx)
 {
 	struct ddsp_vtimelocal_fail *skel;
diff --git a/tools/testing/selftests/scx/scx_test.h b/tools/testing/selftests/scx/scx_test.h
index e402031c2..4b70bf75f 100644
--- a/tools/testing/selftests/scx/scx_test.h
+++ b/tools/testing/selftests/scx/scx_test.h
@@ -17,6 +17,19 @@ enum scx_test_status {
 	SCX_TEST_FAIL,
 };
 
+/* Copied from include/linux/sched/ext.h */
+enum scx_test_exit_kind {
+        SCX_EXIT_NONE,
+        SCX_EXIT_DONE,
+
+        SCX_EXIT_UNREG = 64,    /* BPF unregistration */
+        SCX_EXIT_SYSRQ,         /* requested by 'S' sysrq */
+
+	SCX_EXIT_ERROR = 1024,  /* runtime error, error msg contains details */
+	SCX_EXIT_ERROR_BPF,     /* ERROR but triggered through scx_bpf_error() */
+	SCX_EXIT_ERROR_STALL,   /* watchdog detected stalled runnable tasks */
+};
+
 struct scx_test {
 	/**
 	 * name - The name of the testcase.
diff --git a/tools/testing/selftests/scx/select_cpu_dispatch_bad_dsq.c b/tools/testing/selftests/scx/select_cpu_dispatch_bad_dsq.c
index 4b62ff69c..a7b91d58c 100644
--- a/tools/testing/selftests/scx/select_cpu_dispatch_bad_dsq.c
+++ b/tools/testing/selftests/scx/select_cpu_dispatch_bad_dsq.c
@@ -11,8 +11,6 @@
 #include "select_cpu_dispatch_bad_dsq.bpf.skel.h"
 #include "scx_test.h"
 
-#define SCX_EXIT_ERROR 1024
-
 static enum scx_test_status setup(void **ctx)
 {
 	struct select_cpu_dispatch_bad_dsq *skel;
diff --git a/tools/testing/selftests/scx/select_cpu_dispatch_dbl_dsp.c b/tools/testing/selftests/scx/select_cpu_dispatch_dbl_dsp.c
index 86aa2180f..e32b22963 100644
--- a/tools/testing/selftests/scx/select_cpu_dispatch_dbl_dsp.c
+++ b/tools/testing/selftests/scx/select_cpu_dispatch_dbl_dsp.c
@@ -11,8 +11,6 @@
 #include "select_cpu_dispatch_dbl_dsp.bpf.skel.h"
 #include "scx_test.h"
 
-#define SCX_EXIT_ERROR 1024
-
 static enum scx_test_status setup(void **ctx)
 {
 	struct select_cpu_dispatch_dbl_dsp *skel;
-- 
2.43.0.232.ge79552d197

